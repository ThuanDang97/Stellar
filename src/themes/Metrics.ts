// Used via Metrics.baseMargin
const Metrics = {
  // Variables for spacing
  dimensions: {
    base: 15,
    tiny: 2,
    xs: 5,
    sm: 10,
    md: 20,
    xmd: 25,
    lg: 30,
    xlg: 35,
    xl: 40,
    xxl: 60,
    xxxl: 100,
  },

  // Variables width element
  width: {
    base: 40,
    common: 50,
    full: '100%',
    xs: 20,
    sm: 60,
    md: 70,
    xmd: 105,
    lg: 140,
    xl: 300,
    xxl: 330,
    xxxl: 350,
  },

  height: {
    default: 20,
    base: 40,
    tiny: 6,
    sm: 55,
  },

  // Variables for width height images
  images: {
    md: 15,
    base: 40,
    banner: 220,
  },

  // Variables for width height icons
  icons: {
    base: 25,
    default: 20,
    sm: 10,
  },

  // Variables for width height modal
  modal: {
    base: 200,
    default: 350,
    max: 500,
  },

  // Variables border radius
  borderRadius: {
    default: 3,
    base: 20,
    circle: 50,
  },

  // Variables for border with
  borderWidth: {
    default: 1,
    base: 2,
    tiny: 1.5,
    sm: 4,
    md: 5,
    lg: 6,
    xl: 7,
  },

  // Variable for input
  input: {
    base: 300,
    fontWeight: 400,
  },

  // Variable for layout
  layout: {
    default: '100vh',
    maxWidth: 1280,
  },

  // Variable for breakpoints
  breakPoints: {
    xs: 480,
    sm: 576,
    md: 768,
    lg: 992,
    xl: 1200,
    xxl: 1400,
  },

  // Variable for z-index
  zIndex: {
    top: 4,
    high: 3,
    low: 2,
    under: 1,
  },
}

export default Metrics
