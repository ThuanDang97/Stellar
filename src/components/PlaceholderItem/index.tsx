import React, { memo } from 'react'
import ReactPlaceholder from 'react-placeholder'

// Props type
import { PlaceholderItemProps } from '@self-types/components/PlaceholderItem.props'

// Themes
import { theme } from '@themes/index'
import Metrics from '@themes/Metrics'

// Utils
import { pxToRem } from '@utils/theme'

const PlaceholderItem = ({
  type = 'textRow',
  width = Metrics.width.common,
  height = 13,
  mTop = 0,
  children,
}: PlaceholderItemProps) => (
  <ReactPlaceholder
    type={type}
    ready={false}
    color={theme.colors.mercury}
    style={{
      width: pxToRem(width),
      height: pxToRem(height),
      marginTop: pxToRem(mTop),
    }}
  >
    {children}
  </ReactPlaceholder>
)

export default memo(PlaceholderItem)
