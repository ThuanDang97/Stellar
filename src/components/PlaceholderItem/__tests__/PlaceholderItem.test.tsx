import '@testing-library/jest-dom'
import { render } from '@testing-library/react'

// component
import Metrics from '@themes/Metrics'
import PlaceholderItem from '../index'

// themes

describe('PlaceholderItem Component', () => {
  it('renders PlaceholderItem component', () => {
    const { container } = render(
      <PlaceholderItem
        type="rect"
        width={Metrics.width.base}
        height={Metrics.height.base}
      />,
    )
    expect(container).toMatchSnapshot()
  })
})
