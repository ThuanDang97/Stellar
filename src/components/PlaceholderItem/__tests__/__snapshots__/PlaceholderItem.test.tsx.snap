// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlaceholderItem Component renders PlaceholderItem component 1`] = `
<div>
  <div
    class="rect-shape"
    style="background-color: rgb(230, 230, 230); width: 2.5rem; height: 2.5rem; margin-right: 10px; margin-top: 0rem;"
  />
</div>
`;
